subprojects/gradle-guides-plugin/src/main/java/org/gradle/docs/samples/internal/SamplesDocumentationPlugin.java

```diff
@@ -90,8 +98,12 @@ public void apply(Project project) {
 
         // Samples binaries
         // TODO: This could be lazy if we had a way to make the TaskContainer require evaluation
-        FileCollection generatedTests = createGeneratedTests(tasks, objects, layout);
-        extension.getBinaries().withType(SampleExemplarBinary.class).all(binary -> binary.getTestsContent().from(generatedTests));
+        FileCollection generatedSanityCheckTest = createGeneratedTests(tasks, objects, layout);
```

Comment:
Maybe turn this around a bit...

- Have two tasks that generate sanityCheck and "negative" sanityCheck.  We'd have two different FileCollections here.
- Rename `getExplicitSanityCheck` to something like `getExpectedToFail` (default false)
- And then do something like:
```
binary.getTestsContent().from(binary.getExpectedToFail().flatMap(expectedToFail -> expectedToFail ? generatedNegativeTests : generatedTests))
```

Reply:
I know my initial idea was to create negative sanity checks, but then I thought maybe there's a better alternative. How about the client can explicitly declare what the sanity check does. In that case, all negative examples could simply add a `sanityCheck.sample.conf` file with a `expect-failure: true` configuration.